core/services/chainlink/config_dump.go

```diff
@@ -0,0 +1,270 @@
+package chainlink
+
+import (
+	"encoding/csv"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/pelletier/go-toml/v2"
+	ocrnetworking "github.com/smartcontractkit/libocr/networking"
+
+	"github.com/smartcontractkit/chainlink/core/config"
+	"github.com/smartcontractkit/chainlink/core/config/envvar"
+	"github.com/smartcontractkit/chainlink/core/config/parse"
+	tcfg "github.com/smartcontractkit/chainlink/core/config/toml"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/p2pkey"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+func (app *ChainlinkApplication) ConfigDump() (string, error) {
+	var c Config
+
+	if err := c.loadChainsAndNodes(app.Chains); err != nil {
+		return "", err
+	}
+
+	c.loadLegacyEVMEnv()
+
+	c.loadLegacyCoreEnv()
+
+	b, err := toml.Marshal(c)
+	if err != nil {
+		return "", err
+	}
+	//TODO what about secrets?
+
+	return string(b), nil
+}
+
+//TODO doc
+func (c *Config) loadChainsAndNodes(chains Chains) error {
+	//TODO copy chains and nodes from database
+	//TODO must iterate over pure ORM methods for all chains and nodes
+	//TODO even disabled chains?
+	c.EVM = nil
+	c.Solana = nil
+	c.Terra = nil
+	return nil
+}
+
+//TODO doc
+func (c *Config) loadLegacyEVMEnv() {
+	//TODO read legacy ETH/EVM global overrides, clobbering persisted values already set
+	//TODO ignore EVM_NODES? since DB already updated at this point?
+}
+
+// loadLegacyCoreEnv loads CoreConfig values from legacy environment variables.
+//TODO test
+func (c *Config) loadLegacyCoreEnv() {
+	c.Dev = envvar.NewBool("Dev").ParsePtr()
```

Comment:
Note that this will write even default values directly into the exported config.

I think it would be desirable to omit values if they are not explicitly set, allowing to change defaults in future.

Reply:
Can you elaborate? I set these up to go directly to the env only, so that we do not pick up the default values at all.

Reply:
`ParsePtr()` _does not_ contain the fallback logic from the other methods. It was added for this conversion only, and returns nil instead of defaults.